python/paddle/v2/fluid/layers.py

```diff
@@ -1838,3 +1843,209 @@ def __call__(self):
                     main_program=self.helper.main_program,
                     startup_program=self.helper.startup_program))
         return rlist
+
+
+class DynamicRNN(object):
+    BEFORE_RNN = 0
+    IN_RNN = 1
+    AFTER_RNN = 2
+
+    def __init__(self, name=None, main_program=None, startup_program=None):
+        self.helper = LayerHelper(
+            'dynamic_rnn',
+            name=name,
+            main_program=main_program,
+            startup_program=startup_program)
+        self.status = DynamicRNN.BEFORE_RNN
+        self.lod_rank_table = None
+        self.max_seq_len = None
+        self.step_idx = None
+        self.zero_idx = fill_constant(shape=[1], value=0, dtype='int64')
+        self.mem_dict = dict()
+        self.output_array = []
+        self.outputs = []
+        self.cond = self.helper.create_tmp_variable(dtype='bool')
+        self.cond.stop_gradient = False
```

Comment:
I am little confused about these two concepts, stop_gradient and no_gradient. Are these two the same?

Reply:
`no_grad_set` is a parameter of `backward(...)` method. stop_gradient is an attribute of Python Variable.

The variables whose stop_gradient=True will be passed to `backward(no_grad_set=...)`